### PR TITLE
Feature/base64 nowhitespacecheck

### PR DESCRIPTION
**Pull request by hroest**
- add skipXMLCheck option which will skip the `removeWhitespaces` call. 
- This gives about 10 % speed gain in a critical section of the code and can be turned on/off by the user

### CODE REVIEW

**Comment by hroest** [comment]
note that turning this on is _slightly_ hacky as RFC 2045 on which `xs:base64binary` is based actually allows (or even mandates) line breaks every 76 characters:

https://www.ietf.org/rfc/rfc2045.txt

```
   The encoded output stream must be represented in lines of no more
   than 76 characters each.  All line breaks or other characters not
   found in Table 1 must be ignored by decoding software.  In base64
   data, characters other than those in Table 1, line breaks, and other
   white space probably indicate a transmission error, about which a
   warning message or even a message rejection might be appropriate
   under some circumstances.
```

on the other hand we have other issues than strictly adhering to RFCs and currently nobody adds linebreaks to Base64 texts in XML. Still the current pull request only adds the option but each tool can decide to turn it on by itself

**Comment by hroest** [comment]
maybe in the future we can have a command line parameter that can override the "OpenMS AI" which tries to do automatic peak detection and automatically skips Base64 checks like this but the default would be to turn off these optimizations and automatic checks. I still prefer if there were a way for the user to turn this off

File: src/openms/source/FORMAT/HANDLERS/MzMLHandlerHelper.cpp [hunk 1]
```diff
@@ -161,7 +161,10 @@ namespace OpenMS
     {
       //remove whitespaces from binary data
       //this should not be necessary, but linebreaks inside the base64 data are unfortunately no exception
-      data_[i].base64.removeWhitespaces();
+      // NOTE: this may take up to 10% of reading time with a single thread,
+      //       either omit it or make it more efficient
+      if (!skipXMLCheck)
+        data_[i].base64.removeWhitespaces();
```
Comment (cbielow) [inline src/openms/source/FORMAT/HANDLERS/MzMLHandlerHelper.cpp]: coding convention
```{
  ...
}```

Comment (hroest) [inline src/openms/source/FORMAT/HANDLERS/MzMLHandlerHelper.cpp]: fixed

**Comment by timosachsenberg** [comment]
+1

**Pull request merged**